### PR TITLE
[!feat] Refactor Bybit

### PR DESCRIPTION
**Pull request by colmazia**
## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [ ] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG.md`
- [ ] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the GitHub PR explorer)

### CODE REVIEW

File: bothan-bybit/src/api/types.rs [hunk 1]
```diff
@@ -1,26 +1,53 @@
...
+#[serde(rename_all = "snake_case")]
+pub struct PublicMessageResponse {
+    pub success: bool,
     pub ret_msg: String,
-    /// The result of the response.
-    pub result: T,
-    /// The timestamp of the response.
-    pub time: usize,
+    pub conn_id: String,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
```
Comment (warittornc) [inline bothan-bybit/src/api/types.rs:23]: why do we need to explicitly set serialize for the response?

File: bothan-bybit/src/api/types.rs [hunk 2]
```diff
@@ -1,26 +1,53 @@
...
+pub struct Ticker {
+    pub symbol: String,
+    pub last_price: String,
+    pub high_price24h: String,
+    pub low_price24h: String,
+    pub prev_price24h: String,
+    pub volume24h: String,
+    pub turnover24h: String,
+    pub price24h_pcnt: String,
+    pub usd_index_price: String,
```
Comment (warittornc) [inline bothan-bybit/src/api/types.rs:52]: why is there a fix of snake_case and camelCase for bybit?

File: bothan-bybit/Cargo.toml [hunk 3]
```diff
@@ -7,18 +7,19 @@ license.workspace = true
...
-reqwest = { workspace = true }
+chrono = { workspace = true }
+rust_decimal = { workspace = true }
 serde = { workspace = true }
 serde_json = { workspace = true }
 thiserror = { workspace = true }
 tokio = { workspace = true }
+tokio-tungstenite = { workspace = true }
 tracing = { workspace = true }
 tracing-subscriber = { workspace = true }
```
[the remark this answers is not in the document]
Comment (colmazia) [reply bothan-bybit/Cargo.toml:22]: its formatted

File: bothan-bybit/src/api/types.rs (hunk 2 above)
Comment (colmazia) [reply bothan-bybit/src/api/types.rs:52]: yep its suck but bybit use both snake case and camel case

File: bothan-bybit/src/api/types.rs (hunk 1 above)
Comment (colmazia) [reply bothan-bybit/src/api/types.rs]: deleted

**Review by RogerKSI — APPROVED** [review]

**Pull request merged**